### DISCUSSION TITLE
Seeking Assistance: React Native Storybook - No addons loaded

### DISCUSSION THREAD

**Discussion by Harinony** [storybookjs/storybook#24863] [Help] [answered]
I've implemented Storybook in my React Native project, and everything seems to be working fine except for the addons – they're not loading. Could someone assist me in resolving this issue and getting the addons to load properly?
Here is my code: 
```
import { ComponentMeta, ComponentStory } from '@storybook/react-native';

import { Text } from './Text';

const TextMeta: ComponentMeta<typeof Text> = {
  title: 'Text',
  component: Text,
  args: {
    color: 'red',
  },
  argTypes: {
    color: {
      options: ['red', 'blue', 'green'],
      control: { type: 'radio' },
    },
  },
};

export default TextMeta;

type TextStory = ComponentStory<typeof Text>;

export const Primary: TextStory = args => <Text {...args}> Hello World </Text>;
```

and this is how it looks 
![CleanShot 2023-11-16 at 11 44 00](https://github.com/storybookjs/storybook/assets/31614291/248de933-e7e1-4352-88d3-7e867ebbe685)

**Comment by joevaugh4n** [comment]
Hey @Harinony! Just to say that I'm sharing this with our RN maintainer. Hold tight and hopefully we can get back to you very soon! Thanks for formatting the thread so nicely ❤️

**Comment by dannyhw** [comment]
Please include your main.js config and package.json if possible and I'll be happy to give some advice.

Please also include any error logs that you have

**Reply by dannyhw** [reply, marked as answer]
if this is the same one from discord, I'll repeat my response here just in case

![image](https://github.com/storybookjs/storybook/assets/3481514/5d5d2f0b-f6e7-44ab-bf73-0a5d8a7a4387)

you are mixing addons for react native with addons for web, that won't work
for react native only addon-ondevice addons should be included
you should have a separate config file for web if you want to use web addons

**Comment by arekkubaczkowski** [comment]
@dannyhw I having the same issue, I've tried to provide mobile-only addons and it still shows that "No addons loaded"
any other ideas?


```

const config = {
  stories: ['../src/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
  addons: [
    // '@storybook/addon-essentials',
    // '@storybook/addon-interactions',
    '@storybook/addon-ondevice-actions',
    '@storybook/addon-ondevice-backgrounds',
    '@storybook/addon-ondevice-controls',
    '@storybook/addon-ondevice-notes',
    // '@storybook/addon-actions',
    // '@storybook/addon-react-native-web',
    // '@storybook/addon-actions',
  ],
  framework: {
    name: '@storybook/react-native',
    options: {},
  },
};

export default config;
```

```
        "@storybook/addon-ondevice-actions": "^6.5.3",
        "@storybook/addon-ondevice-backgrounds": "^6.5.3",
        "@storybook/addon-ondevice-controls": "^6.5.3",
        "@storybook/addon-ondevice-notes": "^6.5.3",
```

**Reply by dannyhw** [reply]
Make sure you regenerate the requires file after changing your addons config

**Reply by arekkubaczkowski** [reply]
it seems like I am missing some important part because I have no any requires file, can you elaborate? I am using NX monorepo and I have all the files generated by it

**Reply by dannyhw** [reply]
The nx setup is not updated and doesn't work properly as far as I know. I tried to reach out about it but made no progress 🤷‍♂️ .

**Reply by arekkubaczkowski** [reply]
that's a bad news :( 
I found that out and added it manually:
```
import '@storybook/addon-ondevice-notes/register';
import '@storybook/addon-ondevice-backgrounds/register';
import '@storybook/addon-ondevice-actions/register';
```
however addon-ondevice-actions introduced another error:

Unknown named module: @storybook/global, I tried to install it explicitly but no luck

**Reply by dannyhw** [reply]
this bug depends on your versions if you are on 6.5 make every storybook version 6.5 for example. You could update everything to v7 too.

**Reply by arekkubaczkowski** [reply]
okay I have managed to get that works, thanks a lot!

**Reply by vinayaksaubhri** [reply]
hey @arekkubaczkowski I have the same error and I am using nx too and addons are not working so how did you manage to solve it.
